src/compiler.rs

```diff
@@ -342,9 +349,11 @@ impl<'template> TemplateCompiler<'template> {
     /// (including the expected_close characters), or return an error message if we reach the end
     /// of a line of text without finding it.
     fn consume_tag(&mut self, expected_close: &str) -> Result<&'template str> {
+        let start_len = expected_close.len();

```

Comment:
This confused me until I read the commit message that explained that we're assuming the start of the tag is the same length as the end of the tag. Could you add a comment here to clarify that?